userspace/libscap/scap_bpf.c

```diff
@@ -622,12 +623,36 @@ static int32_t load_bpf_file(scap_t *handle, const char *path)
 				goto cleanup;
 			}
 		}
-		else if(strcmp(shname, "probe_version") == 0) {
-			if(strcmp(PROBE_VERSION, data->d_buf))
+		else if(strcmp(shname, "probe_api_version") == 0) {
+			uint64_t bpf_api_version;
+			got_api_version = true;
+			memcpy(&bpf_api_version, data->d_buf, sizeof(bpf_api_version));
+
+			switch(check_api_compatibility(bpf_api_version))
```

Comment:
What about storing probe_api_version in scap handle and then make a single check for both kmod and eBPF probe in `scap_open_live_int`?

While it looks ok anyway, i fear duplicated checking code! :)

Reply:
Sure. I might as well add an accessor to read the api version from the consumer while I'm at it :)

Reply:
Ultimately I decided against exposing the API version until somebody actually needs it. This feels like an implementation detail of the library.

Still, I'm open to having my mind changed ;)

Reply:
Mmh what about Falco starting message
`Driver API version: X.Y.Z`
?

Reply:
okay.jpg ;)